### DISCUSSION TITLE
insensitivise Maps - why? - need case-Sensitivity

### DISCUSSION THREAD

**Discussion by PeterIttner** [spf13/viper#1288] [Q&A] [answered]
Hello,

I'm using ` viper.MergeConfig(...)` functionality to merge my viper-config with an existing yaml-config-file.
Internally viper reads the config-file and calls

```
...
insensitiviseMap(cfg)
...
```
([viper.go L1583](https://github.com/spf13/viper/blob/97664ba0202811082423479584cc9542c702d69a/viper.go#L1583))

This behaviour removes case-sensitve config values and calls `lower := strings.ToLower(key)` internally.

**I'm asking myself what the reason for this might be.**
I would need my config-values in a case-sensitve way, so I might not be able to use the viper-config-feature for my scenario ...
I would prefer, if this would be configurable.

**Comment by sagikazarmark** [comment, marked as answer]
Case sensitivity has long been requested, but because of the nature of how Viper is built, the decision was made early on that keys should be case insensitive. Ultimately, this proved to be a good decision for many reasons.

There were multiple attempts to make this behavior configurable, but they all failed, because the current architectural structure of Viper.

You can read more about this topic #1014

tl;dr

This is not configurable for now and that's not going to change in the forseeable future. Sorry if that means that Viper is not a good fit for your use case.

**Comment by sagikazarmark** [comment]
Case sensitivity has long been requested, but because of the nature of how Viper is built, the decision was made early on that keys should be case insensitive. Ultimately, this proved to be a good decision for many reasons.

There were multiple attempts to make this behavior configurable, but they all failed, because the current architectural structure of Viper.

You can read more about this topic #1014

tl;dr

This is not configurable for now and that's not going to change in the forseeable future. Sorry if that means that Viper is not a good fit for your use case.
